src/utils/Logger.ts

```diff
@@ -146,15 +148,62 @@ class Logger {
     return error
   }
 
-  getReactNativeLogsFilePath = () => {
-    return RNFS.CachesDirectoryPath + '/rn_logs.txt'
+  getReactNativeLogFilePath = () => {
+    return `${this.getReactNativeLogsDir()}/${format(new Date(), 'yyyyMMdd')}.txt`
   }
 
-  getLogs = async () => {
-    // TODO(Rossy) Does this technique of passing logs back as a string
-    // fail when the logs get too big?
+  getReactNativeLogsDir = () => {
+    return `${RNFS.CachesDirectoryPath}/rn_logs`
+  }
+
+  getCombinedLogsFilePath = () => {
+    // For now we need to export to a world-readable directory on Android
+    // TODO: use the FileProvider approach so we don't need to do this.
+    // See https://developer.android.com/reference/androidx/core/content/FileProvider
+    // and https://github.com/chirag04/react-native-mail/blame/340618e4ef7f21a29d739d4180c2a267a14093d3/android/src/main/java/com/chirag/RNMail/RNMailModule.java#L106
+    const path = Platform.OS === 'ios' ? RNFS.TemporaryDirectoryPath : RNFS.ExternalDirectoryPath
+    return `${path}/rn_logs.txt`
+  }
+
+  cleanupOldLogs = async () => {
+    try {
+      const logFileCombinedPath = this.getCombinedLogsFilePath()
+      const logDir = this.getReactNativeLogsDir()
+
+      // If legacy log file or combined logs exist, delete it
+      if (await RNFS.exists(logFileCombinedPath)) {
+        await RNFS.unlink(logFileCombinedPath)
+      }
+
+      // Get the list of log files
+      const logFiles = await RNFS.readDir(logDir)
+
+      // Delete log files older than 28 days
+      if (logFiles.length > 0) {
+        logFiles.forEach((file) => {
+          RNFS.stat(file.path)
```

Comment:
why not use `await` here?

Reply:
Changed in https://github.com/valora-inc/wallet/commit/f8b84262e339c90deeee4cbaad1d5bf7742f9960!